deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,188 @@
+mkdir -p "$SCRIPT_DIR"
+INSTALLED_SPLUNK_OTEL_VERSION=""
+
+if [ -f $SPLUNK_OTEL_BINARY_FILE ]; then
+  # Output of `otelcol --version` is of the form:
+  # otelcol version vX.X.X
+  # Capture output of `otelcol --version` as an array, then extract version
+  INSTALLED_SPLUNK_OTEL_VERSION=($($SPLUNK_OTEL_BINARY_FILE --version))
+  INSTALLED_SPLUNK_OTEL_VERSION=${INSTALLED_SPLUNK_OTEL_VERSION[2]}
+fi
+
+if [ ! -f $SPLUNK_OTEL_BINARY_FILE ] || [ $INSTALLED_SPLUNK_OTEL_VERSION != $SPLUNK_OTEL_VERSION ]; then
+  # If the binary is already installed it's the wrong version and it needs to be removed before downloading again
+  # to avoid "Text file busy" errors
+  if [ -f $SPLUNK_OTEL_BINARY_FILE ]; then
+    rm "$SPLUNK_OTEL_BINARY_FILE"
+  fi
+
+  # Download Splunk's distribution of the OpenTelemetry Collector
+  curl --output "$SPLUNK_OTEL_BINARY_FILE" $SPLUNK_OTEL_BINARY_DOWNLOAD_URL || { echo "Failed to download $OTEL_BINARY_DOWNLOAD_URL"; exit 1; }
+  chmod +x "$SPLUNK_OTEL_BINARY_FILE"
+else
+  echo "Splunk OpenTelemetry Collector '${SPLUNK_OTEL_VERSION}' is already installed"
+fi
+
+# The Spark receiver should only be run in one instance per-Cluster. Run
+# it on the driver node, as there's one per-cluster.
+# More info on Databricks init script environment variables:
+# https://docs.databricks.com/en/init-scripts/environment-variables.html#use-secrets-in-init-scripts
+if [ $DB_IS_DRIVER = "TRUE" ]; then
+        OPTIONAL_SPARK_RECEIVER=", apachespark"
+else
+        OPTIONAL_SPARK_RECEIVER=""
+fi
+
+collector_config="
+extensions:
+  bearertokenauth:
+    token: $DATABRICKS_ACCESS_TOKEN
+
+receivers:
+  apachespark:
+    # https://community.databricks.com/t5/data-engineering/how-to-obtain-the-server-url-for-using-spark-s-rest-api/td-p/83410
+    endpoint: https://$DATABRICKS_CLUSTER_HOSTNAME/driver-proxy-api/o/0/$DB_CLUSTER_ID/40001
+    auth:
+      authenticator: bearertokenauth
+  # TODO: Identify any additional scrapers that are necessary and useful
+  hostmetrics:
+    scrapers:
+      cpu:
+      memory:
+      network:
+
+processors:
+  batch:
+    send_batch_size: 10000
+    timeout: 10s
+  resourcedetection:
+    detectors: [system]
+  resource:
+    attributes:
+      - key: databricks.cluster.name
+        value: \"$DB_CLUSTER_NAME\"
+        action: upsert
+      - key: databricks.cluster.id
+        value: \"$DB_CLUSTER_ID\"
+        action: upsert
+      - key: databricks.node.driver
+        value: \"$DB_IS_DRIVER\"
+        action: upsert
+
+exporters:
+  signalfx:
+    access_token: $SPLUNK_ACCESS_TOKEN
+    realm: ${SPLUNK_REALM:-us0}
+
+service:
+  extensions: [bearertokenauth]
+  pipelines:
+    metrics:
+      receivers: [hostmetrics$OPTIONAL_SPARK_RECEIVER]
+      processors: [batch, resourcedetection, resource]
+      exporters: [signalfx]
+"
+
+echo "$collector_config" > "$CONFIG_FILE"
+
+collector_service="
```

Comment:
Did you get a chance to try installing the collector with a package manager on databricks? I am just wondering if these nodes are fixed os type, then maybe we could install with a package manager. Looking at the service file [here](https://github.com/signalfx/splunk-otel-collector/blob/main/packaging/fpm/splunk-otel-collector.service), we'd get stuff like dedicated user/group and a better experience in general.